e2e/helpers/action.ts

```diff
@@ -84,7 +84,7 @@ export class ActionStack {
     console.log("--------------------------------");
     console.log(TIME_KEY, "starting.");
 
-    let action: Action<unknown> | undefined;
+    let action: Action<unknown> | null | undefined = null;
     while ((action = this.actions.pop())) {
```

Comment:
```suggestion
    let action: Action<unknown> | null = null;
    while ((action = this.actions.pop() ?? null)) {
```

I don't think we need `undefined` and `null` both here in the type definition.

Reply:
Thanks, done.